Release v1.2.2 (#5068)
grafana/k6: release notes/v1.2.2.md

> k6 1.2.2 is a small patch release fixing a panic.

Author: ```suggestion
k6 1.2.2 is a small patch release fixing a panic and two other smaller bugfixes.
```

> k6 1.2.2 is here 🎉! This is small patch release fixing a panic.

Reviewer: ```suggestion
k6 1.2.2 is a small patch release fixing a panic.
```
it doesn't sound something to be too excited 😆

## Bug fixes

> - [#5067](https://github.com/grafana/k6/pull/5067) fix: do not panic on `k6 login cloud`.

Reviewer: ```suggestion
- [#5067](https://github.com/grafana/k6/pull/5067) fixes a panic on the deprecated `k6 login cloud` command. Thanks @indygriffiths for reporting it!
```

* Keeps the same style as we do for other releases
* Thanks the user for reporting
* mention that the command is the deprecated one so they don't confuse it with the official command.

> 

Author: ```suggestion
- [#5069](https://github.com/grafana/k6/pull/5069) Fixes group order in end of test summary when scenarios are used. 
- [#5070](https://github.com/grafana/k6/pull/5070) Adds nullish check to the new getByRole and add tests for other getBy* APIs nullish checks.
```
